### PR TITLE
jetson-agx-orin: Add support for Jetson AGX Orin

### PR DESCRIPTION
**Pull request by robertogs2**
This adds support for tegra234 Jetson Orin

Signed-off-by: Roberto Gutierrez roberto.gutierrez@ridgerun.com

### CODE REVIEW

**Comment by ichergui** [comment]
Hey @robertogs2 
Thanks for the PR.
FYI, I started reviewing the PR and doing some testing.
Will keep you updated

**Comment by ichergui** [comment]
@robertogs2

**Comment by ichergui** [comment]
@robertogs2 
I installed `tegra-firmware` manually into my `Jetson-orin-agx` which fixes `nvpmodel.service` 
The `NVIDIA Argus daemon` systemd service still failing
Here is log from `journald`

```
root@jetson-agx-orin-devkit:~# journalctl -u nvargus-daemon --no-pager 
Apr 19 01:53:18 jetson-agx-orin-devkit systemd[1]: Started NVIDIA Argus daemon.
Apr 19 01:53:18 jetson-agx-orin-devkit nvargus-daemon[522]: Opening channel /dev/nvhost-vic failed
Apr 19 01:53:18 jetson-agx-orin-devkit nvargus-daemon[522]: Couldn't create ddkvic Session: Cannot allocate memory
Apr 19 01:53:18 jetson-agx-orin-devkit nvargus-daemon[522]: nvbuf_utils: Could not create Default NvBufferSession
Apr 19 01:53:18 jetson-agx-orin-devkit systemd[1]: nvargus-daemon.service: Main process exited, code=exited, status=255/EXCEPTION
Apr 19 01:53:18 jetson-agx-orin-devkit systemd[1]: nvargus-daemon.service: Failed with result 'exit-code'.
```

**Review by ichergui — CHANGES_REQUESTED** [review]
Hey @robertogs2 
Could you please apply the following patch to fix FW dependency for `tegra234` ? 
[0001-tegra-firmware-fix-FWDEPS-for-tegra234.patch.txt](https://github.com/OE4T/meta-tegra/files/8512228/0001-tegra-firmware-fix-FWDEPS-for-tegra234.patch.txt)

I'm not sure if we need to update `tegra234` xusb firmware.
I will keep you updated.

File: recipes-bsp/tegra-binaries/tegra-helper-scripts-native_34.1.0.bb [hunk 1]
```diff
@@ -7,7 +7,7 @@ INHIBIT_DEFAULT_DEPS = "1"
 inherit native
 
 SRC_URI = " \
-    file://tegra194-flash-helper.sh \
+    file://${SOC_FAMILY}-flash-helper.sh \
```
Comment (madisongh) [inline recipes-bsp/tegra-binaries/tegra-helper-scripts-native_34.1.0.bb:10]: Instead of this change, just add `tegra234-flash-helper.sh` to the SRC_URI.  Take a look at the version of this recipe in the `master` branch as an example - it installs all of the helper scripts for all chip variants.

File: recipes-bsp/tegra-binaries/tegra-helper-scripts-native_34.1.0.bb [hunk 2]
```diff
@@ -16,9 +16,9 @@ SRC_URI = " \
...
-    install -d ${D}${bindir}/tegra194-flash
-    install -m 0755 ${S}/tegra194-flash-helper.sh ${D}${bindir}/tegra194-flash/
-    install -m 0755 ${S}/nvflashxmlparse.py ${D}${bindir}/tegra194-flash/nvflashxmlparse
-    install -m 0755 ${S}/make-sdcard.sh ${D}${bindir}/tegra194-flash/make-sdcard
-    install -m 0755 ${S}/tegra-signimage-helper.sh ${D}${bindir}/tegra194-flash/tegra-signimage-helper
+    install -d ${D}${bindir}/${SOC_FAMILY}-flash
+    install -m 0755 ${S}/${SOC_FAMILY}-flash-helper.sh ${D}${bindir}/${SOC_FAMILY}-flash/
+    install -m 0755 ${S}/nvflashxmlparse.py ${D}${bindir}/${SOC_FAMILY}-flash/nvflashxmlparse
+    install -m 0755 ${S}/make-sdcard.sh ${D}${bindir}/${SOC_FAMILY}-flash/make-sdcard
+    install -m 0755 ${S}/tegra-signimage-helper.sh ${D}${bindir}/${SOC_FAMILY}-flash/tegra-signimage-helper
```
Comment (madisongh) [inline recipes-bsp/tegra-binaries/tegra-helper-scripts-native_34.1.0.bb:23]: See above

File: recipes-bsp/tegra-binaries/tegra-nvphs-base_34.1.0.bb [hunk 3]
```diff
@@ -15,6 +15,7 @@ SRC_URI[init.sha256sum] = "${INITSUM}"
 NVPHSD_MACHINE_CONF = "nvphsd.conf"
 NVPHSD_MACHINE_CONF:tegra186 = "nvphsd.conf.t186"
 NVPHSD_MACHINE_CONF:tegra194 = "nvphsd.conf.t194"
+NVPHSD_MACHINE_CONF:tegra234 = "nvphsd_common.conf"
```
Comment (madisongh) [inline recipes-bsp/tegra-binaries/tegra-nvphs-base_34.1.0.bb:18]: Looks like nvphsd doesn't need this config renaming stuff any more, and we should just install both of the config files into `${D}${sysconfdir}` as-is, for all platforms.

File: recipes-graphics/l4t-graphics-demos/l4t-graphics-demos.inc [hunk 4]
```diff
@@ -2,7 +2,7 @@ HOMEPAGE = "https://developer.nvidia.com/embedded"
 
 COMPATIBLE_MACHINE = "(tegra)"
 
-L4T_DEB_SOCNAME = "t194"
+L4T_DEB_SOCNAME ?= "t194"
```
Comment (madisongh) [inline recipes-graphics/l4t-graphics-demos/l4t-graphics-demos.inc:5]: No, this should be straight assignment, so we're always fetching the `t194` version.  The content doesn't actually vary between the two, so we don't need to account for the SoC name.

File: recipes-multimedia/argus/tegra-mmapi-34.1.0.inc [hunk 5]
```diff
@@ -1,7 +1,7 @@
 HOMEPAGE = "http://developer.nvidia.com"
 LICENSE = "Proprietary & BSD-3-Clause"
 
-L4T_DEB_SOCNAME = "t194"
+L4T_DEB_SOCNAME ?= "t194"
```
Comment (madisongh) [inline recipes-multimedia/argus/tegra-mmapi-34.1.0.inc:4]: Same comment as for l4t-graphics-demos -- please revert this change.

**Review by madisongh — CHANGES_REQUESTED** [review]
Generally looks good, just a few changes needed - see inline.

Thanks very much!

**Comment by robertogs2** [comment]
Hey @ichergui @madisongh,
I applied the patch and the required changes. It seems like with all those modifications both daemons are working correctly now. Please verify that, thank you!

**Review by ichergui — APPROVED** [review]
LGTM

**Review by madisongh — APPROVED** [review]
